app/src/main/java/com/kickstarter/ui/activities/CreatorDashboardActivity.kt

```diff
@@ -0,0 +1,141 @@
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = CreatorDashboardLayoutBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        adapter = CreatorDashboardAdapter(viewModel.inputs)
+
+        binding.creatorDashboardRecyclerView.adapter = adapter
+
+        val layoutManager = LinearLayoutManager(this)
+        binding.creatorDashboardRecyclerView.layoutManager = layoutManager
+
+        // Set up the bottom sheet recycler view.
+        bottomSheetAdapter = CreatorDashboardBottomSheetAdapter(viewModel.inputs)
+        binding.creatorDashboardBottomSheetRecyclerView.adapter = bottomSheetAdapter
+        binding.creatorDashboardBottomSheetRecyclerView.layoutManager = LinearLayoutManager(this)
+
+        bottomSheetBehavior = BottomSheetBehavior.from(binding.creatorDashboardBottomSheetRecyclerView)
+
+        fadeAndTranslateToolbarTitleOnExpand(
+            binding.creatorDashboardToolbarLayout.creatorDashboardAppBar,
+            binding.creatorDashboardToolbarLayout.creatorDashboardProjectNameSmall
+        )
+
+        viewModel.outputs.bottomSheetShouldExpand()
+            .compose(bindToLifecycle())
+            .compose(Transformers.observeForUI())
+            .subscribe { bottomSheetShouldExpand(it) }
+
+        viewModel.outputs.progressBarIsVisible()
+            .compose(bindToLifecycle())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe {
+                ViewUtils.setGone(
+                    binding.creatorDashboardProgressBar, !it
+                )
+            }
+
+        viewModel.outputs.projectDashboardData()
+            .compose(bindToLifecycle())
+            .compose(Transformers.observeForUI())
+            .subscribe { adapter.takeProjectDashboardData(it) }
+
+        viewModel.outputs.projectName()
+            .compose(bindToLifecycle())
+            .compose(Transformers.observeForUI())
+            .subscribe { setProjectNameTextViews(it) }
+
+        viewModel.outputs.projectsForBottomSheet()
+            .compose(bindToLifecycle())
+            .compose(Transformers.observeForUI())
+            .subscribe { setProjectsForDropdown(it) }
+
+        createAndSetBottomSheetCallback()
+
+        binding.creatorDashboardBottomSheetScrim.setOnClickListener { bottomSheetScrimClicked() }
+    }
+
+    override fun back() {
+        if (bottomSheetBehavior.state == BottomSheetBehavior.STATE_EXPANDED) {
+            viewModel.inputs.backClicked()
+        } else {
+            super.back()
+        }
+    }
+
+    private fun bottomSheetScrimClicked() {
+        viewModel.inputs.scrimClicked()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        binding.creatorDashboardRecyclerView.adapter = null
+        binding.creatorDashboardBottomSheetRecyclerView.adapter = null
+        bottomSheetBehavior.setBottomSheetCallback(null)
```

Comment:
`setBottomSheetCallback()` is deprecated. Passing a null here just logs a statement to the console that this method is deprecated, and clears the callbacks but does nothing else, is that the intended purpose? If not i think we can remove this. If so, disregard!

Reply:
Ah now i'm seeing that this is in onDestroy, so it makes sense that we would want to clear the callbacks here. Nevermind!